### PR TITLE
Infer filetype and datatype from filename

### PR DESCRIPTION
**Pull request by pkerpedjiev**
## Description

What was changed in this pull request?

Added functions to infer the filetype and datatype of a dataset from its filename. Lifted directly from https://github.com/higlass/higlass-manage/blob/687b40f56fca3c392f401b294828a8c0f9f66c5c/higlass_manage/common.py#L44 and added a test.

Once this is merged, `higlass-manage` can be updated to use the functions here.

Why is it necessary?

To factor out this functionality so that it can be maintained in a common place.

Fixes #___

## Checklist

- [x] Unit tests added or updated
- [ ] Documentation added or updated
- [ ] Updated CHANGELOG.md
- [x] Ran `black` on the root directory

### CODE REVIEW

**Comment by pkerpedjiev** [comment]
Good idea. Updated!

File: higlass/utils.py [hunk 1]
```diff
@@ -23,3 +25,118 @@ def hg_cmap(cmap_name, resolution=256, reverse=False):
...
+    if op.splitext(filename)[1] == ".bedpe":
+        return "bedpe"
+
+
+def recommend_datatype(filetype):
+    if filetype == "bedfile":
+        return "bedlike"
+
+
+filetypes = {
```
Comment (nvictus) [inline higlass/utils.py]: I'd capitalize this variable

File: higlass/utils.py [hunk 2]
```diff
@@ -23,3 +25,118 @@ def hg_cmap(cmap_name, resolution=256, reverse=False):
...
+                file=sys.stderr,
+            )
+            if recommended_datatype is not None:
+                print(
+                    "Based on the filetype, you may want to try the datatype: {}".format(
+                        recommended_datatype
+                    )
+                )
+
+    return (filetype, datatype)
```
Comment (nvictus) [inline higlass/utils.py]: parens are unnecessary

**Comment by pkerpedjiev** [comment]
Done!

**Pull request merged**